src/spetlr/transformers/clean_column_names_transformer.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+
+from pyspark.sql import DataFrame
+
+from spetlr.etl import Transformer
+
+
+class CleanColumnNamesTransformer(Transformer):
+    """
+    Clean up columns names by removing special characters,
+    trimming whitespaces and changing other whitespaces to underscores
+
+    Parameters
+    ----------
+        exclude_columns : Potential columns to exclude
+    """
+
+    def __init__(
+        self,
+        exclude_columns: List[str] = None,
+        dataset_input_keys: List[str] = None,
+        dataset_output_key: str = None,
+        consume_inputs: bool = False,
+    ):
+        super().__init__(
+            dataset_input_keys=dataset_input_keys,
+            dataset_output_key=dataset_output_key,
+            consume_inputs=consume_inputs,
+        )
+        self.exclude_columns = exclude_columns or []
+        self.keep_chars = [" ", "_"]
+        self.replace_chars = [" ", "-"]
```

Comment:
So how i understand it keep_chars are special characters not removed removed, and replace_chars are characters replaced by _. Doesn't that mean '-' is removed by not being in keep_chars instead of replaced by _? I don't see this case in the test.

Reply:
I have tested and updated the logic!